test/local/ip_blocklist_tests.js

```diff
@@ -0,0 +1,207 @@
+/* Any copyright is dedicated to the Public Domain.
+ * http://creativecommons.org/publicdomain/zero/1.0/ */
+
+require('ass')
+var test = require('tap').test
+
+var log = {
+  info: function () {},
+  error: function () {},
+  trace: function () {}
+}
+
+var config = {
+  ipBlocklist: {
+    updatePollInterval: 1, // 1 Second
+    batchSize: 10000, // 1 Second
+    batchDelayMs: 1 // 1 Second
+  }
+}
+
+var IPBlocklist = require('../../lib/ip_blocklist')(log, config)
+var IPBlocklistManager = require('../../lib/ip_blocklist_manager')(log, config)
+
+var lists = [
+  './test/mocks/firehol_level1.netset',
+  './test/mocks/simple.netset'
+]
+
+var commonTestCases = [
+  {
+    name: 'IPBlocklist',
+    blocklistClass: IPBlocklist,
+    list: lists[0]
+  },
+  {
+    name: 'IPBlocklistManager',
+    blocklistClass: IPBlocklistManager,
+    list: lists
+  }
+]
+
+commonTestCases.forEach(function (testCase) {
+  var BlocklistClass = testCase.blocklistClass
+  var filePath = testCase.list
+  var name = testCase.name
+
+  test(
+    name + ', calling contains without loading csv return false',
+    function (t) {
+      var ipBlocklist = new BlocklistClass()
+      var result = ipBlocklist.contains('0.0.0.0')
+      t.equal(result, false, 'should not have found ip')
+      t.end()
+    }
+  )
+
+
+  test(
+    name + ', load ip blocklist',
+    function (t) {
+      var ipBlocklist = new BlocklistClass()
+
+      ipBlocklist.load(filePath)
+        .then(function () {
+          t.end()
+        })
+        .catch(function () {
+          t.fail('Failed to load csv')
+          t.end()
+        })
+    }
+  )
+
+  test(
+    name + ', returns true for empty ip',
+    function (t) {
+      var ipBlocklist = new BlocklistClass()
+
+      ipBlocklist.load(filePath)
+        .then(function () {
+          var result = ipBlocklist.contains()
+          t.equal(result, true, 'return true for empty ip')
+          t.end()
+        })
+    }
+  )
+
+  test(
+    name + ', returns true for invalid ip',
+    function (t) {
+      var ipBlocklist = new BlocklistClass()
+
+      ipBlocklist.load(filePath)
+        .then(function () {
+          var result = ipBlocklist.contains('notip')
+          t.equal(result, true, 'return true for invalid ip')
```

Comment:
I think this is fair, the default is to block on [server-side errors](https://github.com/mozilla/fxa-customs-server/blob/200d7e97a88a478430ea59723886d48786205b00/lib/server.js#L186). I can rethrow the error and have the server handle it.